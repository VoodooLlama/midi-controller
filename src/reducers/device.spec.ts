import { Input, Output } from 'webmidi';
import deviceReducer from './device';
import {
    DEVICE_ACTIONS,
    SelectInputAction,
    SetInputDevicesAction,
    SetOutputDevicesAction
} from '../actions/device';

let _: any = undefined;

describe('#deviceReducer', () => {
    it('SET_INPUTS', () => {
        let result;

        expect(
            deviceReducer(_, {
                type: DEVICE_ACTIONS.SET_INPUTS,
                devices: [
                    {
                        id: 'deviceId',
                        manufacturer: 'deviceManufacturer',
                        name: 'deviceName',
                        state: 'disconnected',
                        type: 'input'
                    } as Input
                ]
            } as SetInputDevicesAction)
        ).toMatchSnapshot();
    });

    it('SET_OUTPUTS', () => {
        let result;

        expect(
            deviceReducer(_, {
                type: DEVICE_ACTIONS.SET_OUTPUTS,
                devices: [
                    {
                        id: 'deviceId',
                        manufacturer: 'deviceManufacturer',
                        name: 'deviceName',
                        state: 'disconnected',
                        type: 'output'
                    } as Output
                ]
            } as SetOutputDevicesAction)
        ).toMatchSnapshot();
    });

    it('SELECT_INPUT', () => {
        let result;

        expect(
            deviceReducer(_, {
                type: DEVICE_ACTIONS.SELECT_INPUT,
                deviceId: 'deviceId'
            } as SelectInputAction)
        ).toMatchSnapshot();
    });

    it('SELECT_OUTPUT', () => {
        let result;

        expect(
            deviceReducer(_, {
                type: DEVICE_ACTIONS.SELECT_INPUT,
                deviceId: 'deviceId'
            } as SelectInputAction)
        ).toMatchSnapshot();
    });
});
