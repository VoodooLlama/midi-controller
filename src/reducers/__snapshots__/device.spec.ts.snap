// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#deviceReducer SELECT_INPUT 1`] = `
Object {
  "inputDevices": Array [],
  "inputDevicesById": Object {},
  "outputDevices": Array [],
  "outputDevicesById": Object {},
  "selectedInputDeviceId": "deviceId",
  "selectedOutputDeviceId": "",
}
`;

exports[`#deviceReducer SELECT_OUTPUT 1`] = `
Object {
  "inputDevices": Array [],
  "inputDevicesById": Object {},
  "outputDevices": Array [],
  "outputDevicesById": Object {},
  "selectedInputDeviceId": "deviceId",
  "selectedOutputDeviceId": "",
}
`;

exports[`#deviceReducer SET_INPUTS 1`] = `
Object {
  "inputDevices": Array [
    "deviceId",
  ],
  "inputDevicesById": Object {
    "deviceId": Object {
      "id": "deviceId",
      "manufacturer": "deviceManufacturer",
      "name": "deviceName",
      "state": "disconnected",
      "type": "input",
    },
  },
  "outputDevices": Array [],
  "outputDevicesById": Object {},
  "selectedInputDeviceId": "",
  "selectedOutputDeviceId": "",
}
`;

exports[`#deviceReducer SET_OUTPUTS 1`] = `
Object {
  "inputDevices": Array [],
  "inputDevicesById": Object {},
  "outputDevices": Array [
    "deviceId",
  ],
  "outputDevicesById": Object {
    "deviceId": Object {
      "id": "deviceId",
      "manufacturer": "deviceManufacturer",
      "name": "deviceName",
      "state": "disconnected",
      "type": "output",
    },
  },
  "selectedInputDeviceId": "",
  "selectedOutputDeviceId": "",
}
`;
